### ISSUE TITLE
DecoderException upon attempting to join server

### ISSUE THREAD

**Issue by MrKashew** [Sinytra/ForgifiedFabricAPI#40]
### Describe the bug

DecoderException preventing client from joining server

### Steps to reproduce

Context:
- This test is a follow up of a bug I am having on Automodpack:
- This might be a follow up issue once I can replicate on FFAPI
    - https://github.com/Skidamek/AutoModpack/issues/166

Testing:
Modpack I am using on Server and Client for testing:
https://drive.google.com/file/d/1zInmQPAhP0iDzIJ2OJLPQZ6VABdVCUCc/view?usp=sharing
- Running server on the same modpack, attempting to join results in
- ![image](https://github.com/Sinytra/ForgifiedFabricAPI/assets/53996665/b9498bbe-5513-4442-bb0d-f5605806d6d5)

[latest.log](https://github.com/Sinytra/ForgifiedFabricAPI/files/12568072/latest.log)
[debug.log](https://github.com/Sinytra/ForgifiedFabricAPI/files/12568073/debug.log)

Do let me know if there is anything more I can provide or try

**Comment by MrKashew** [comment]
my mod set has changed a bit but can confirm that is happens when joining server but not on singleplayer

**Comment by Su5eD** [comment]
The modpack you've provided contains a large amount of mods. Could you please track down the mod(s) causing the issue and see if it can be reproduced in a clean environment?

**Comment by MrKashew** [comment]
Sure thing, I’ll take a look and narrow it down tonight

**Comment by MrKashew** [comment]
~~removed creativecore mods and rubidium mods and receiving this error on launch - any idea what this log is saying~~

edit: put it in separate issue #59
removing from modpack for now to continue testing for this issue itself

**Comment by MrKashew** [comment]
seems the issue is with a few mods:
either:
- removing just FFAPI solves decoder (so it is caused by an incompatibility)
- removing just Controlling + Searchables (clientside) solves decoder
- removing just NoSeeNoTick **_and_** SmartCompletion solves decoder

however a new instance with only FFAPI and the mods detailed does not decoder
so the issue may be an across interaction between FFAPI + the mods listed + additional mods?
I can look into it more when I have time but it's becoming harder to pinpoint

TLDR these with FFAPI have been confirmed to **_conditionally_** cause decoder:

- https://modrinth.com/mod/controlling (clientside)
    - https://modrinth.com/mod/searchables (dependency)
- https://www.curseforge.com/minecraft/mc-mods/no-see-no-tick
- https://www.curseforge.com/minecraft/mc-mods/smart-completion

**Comment by MrKashew** [comment]
@Su5eD seems the issue is with a few mods, findings are as follows
either:
- removing just FFAPI solves decoder (so it is caused by an incompatibility)
- removing just Controlling + Searchables (clientside) solves decoder
- removing just NoSeeNoTick **_and_** SmartCompletion solves decoder

however a new instance with only FFAPI and the mods detailed does not decoder
so the issue may be an across interaction between FFAPI + the mods listed + additional mods?
I can look into it more when I have time but it's becoming harder to pinpoint

TLDR these with FFAPI have been confirmed to **_conditionally_** cause decoder:

- https://modrinth.com/mod/controlling (clientside)
    - https://modrinth.com/mod/searchables (dependency)
- https://www.curseforge.com/minecraft/mc-mods/no-see-no-tick
- https://www.curseforge.com/minecraft/mc-mods/smart-completion

**Comment by MrKashew** [comment]
Are you using Prism Launcher, I can provide an instance and provide more info

**Comment by Su5eD** [comment]
Yes I am, that could help

**Comment by MrKashew** [comment]
I was unable to replicate this in a clean environment with the mods I've listed above
I did try this: removing mod sets (mod that supports subfolders for mods, for organization etc) and using all the mods and FFAPI
https://modrinth.com/mod/mod-sets

This fixed the issue!
So I suspect it also has to do with mod sets + FFAPI + the mods listed, but I cannot replicate from scratch with these

This means I am no longer hard-blocked I believe however, as I can simply stop using mod sets as it is not a functional mod just a qol mod for modpackers

If you are curious to look into this yourself and find out more I will be happy to help debug more
Here is the Prism Instance, for the server simply use the same mods folder but remove the subfolder "clientside" and leave everything else

https://drive.google.com/file/d/1uMPBrbgPa_hyuREpgS8jStFh6zqJtLw1/view?usp=sharing

**Comment by MrKashew** [comment]
After quite a while I was able to work it down, seems a handful of mods were for some reason causing this incompatibility/decoder exception between server and client that I simply have removed from both. Was not directly related to this mod, closing this issue
